pkg/api/defaults.go

```diff
@@ -424,6 +424,8 @@ func (p *Properties) setMasterProfileDefaults(isUpgrade, isScale bool, cloudName
 		} else if p.MasterProfile.Distro == AKS1804Deprecated {
 			p.MasterProfile.Distro = AKSUbuntu1804
 		}
+	} else if p.OrchestratorProfile.KubernetesConfig.CustomHyperkubeImage != "" && p.MasterProfile.Distro == "" {
```

Comment:
why not just put this in the if statement? I think all conditions above apply... If imageRef is set we don't want to set the distro here either

Reply:
not sure what you mean

Reply:
I mean:

```
if p.MasterProfile.Distro == "" && p.MasterProfile.ImageRef == nil {
		if p.OrchestratorProfile.IsKubernetes() && p.OrchestratorProfile.KubernetesConfig.CustomHyperkubeImage != "" {
			p.MasterProfile.Distro = AKSUbuntu1604
		} else {
			p.MasterProfile.Distro = Ubuntu
		}
	} else if p.OrchestratorProfile.IsKubernetes() && (isUpgrade || isScale) {
		if p.MasterProfile.Distro == AKSDockerEngine || p.MasterProfile.Distro == AKS1604Deprecated {
			p.MasterProfile.Distro = AKSUbuntu1604
		} else if p.MasterProfile.Distro == AKS1804Deprecated {
			p.MasterProfile.Distro = AKSUbuntu1804
		}
	}
```

Reply:
All the unit tests you wrote should pass with this and it's more readable (the 3rd if is redundant otherwise) and it should cover all the logic we want (I think). If not, can you add a unit test case that passes in the current code that doesn't pass with the code above?